src/devices_models/device_constructors/branch_constructor.jl

```diff
@@ -463,30 +471,38 @@ function construct_device!(
     ::ArgumentConstructStage,
     ::DeviceModel{T, F},
     ::NetworkModel{S},
-) where {T <: PSY.DCBranch, F <: HVDCP2PLossless, S <: PM.AbstractPowerModel} end
+) where {
+    T <: TwoTerminalHVDCTypes,
+    F <: HVDCTwoTerminalLossless,
+    S <: PM.AbstractPowerModel,
+} end
 
 function construct_device!(
     container::OptimizationContainer,
     sys::PSY.System,
     ::ModelConstructStage,
     model::DeviceModel{T, U},
     network_model::NetworkModel{S},
-) where {T <: PSY.DCBranch, U <: HVDCP2PLossless, S <: PM.AbstractPowerModel}
+) where {
+    T <: TwoTerminalHVDCTypes,
+    U <: HVDCTwoTerminalLossless,
+    S <: PM.AbstractPowerModel,
+}
     devices =
         get_available_components(T, sys, get_attribute(model, "filter_function"))
     add_constraints!(container, FlowRateConstraint, devices, model, network_model)
     add_constraint_dual!(container, sys, model)
     return
 end
 
-# Repeated method to avoid ambiguity between HVDCP2PUnbounded and HVDCP2PLossless
+# Repeated method to avoid ambiguity between HVDCTwoTerminalUnbounded and HVDCTwoTerminalLossless
```

Comment:
Would it be worth refactoring to make each duplicate function call a common function such that no logic is duplicated?

Reply:
It would still cause the same issue because the arguments will still be ambiguous